core/src/test/java/org/apache/iceberg/util/TestPartitionMap.java

```diff
@@ -0,0 +1,222 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.util;
+
+import static org.apache.iceberg.types.Types.NestedField.required;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.StructLike;
+import org.apache.iceberg.TestHelpers.Row;
+import org.apache.iceberg.data.GenericRecord;
+import org.apache.iceberg.data.Record;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.apache.iceberg.types.Types;
+import org.junit.jupiter.api.Test;
+
+public class TestPartitionMap {
+
+  private static final Schema SCHEMA =
+      new Schema(
+          required(1, "id", Types.IntegerType.get()),
+          required(2, "data", Types.StringType.get()),
+          required(3, "category", Types.StringType.get()));
+  private static final PartitionSpec UNPARTITIONED_SPEC = PartitionSpec.unpartitioned();
+  private static final PartitionSpec BY_DATA_SPEC =
+      PartitionSpec.builderFor(SCHEMA).identity("data").withSpecId(1).build();
+  private static final PartitionSpec BY_DATA_CATEGORY_BUCKET_SPEC =
+      PartitionSpec.builderFor(SCHEMA).identity("data").bucket("category", 8).withSpecId(3).build();
+  private static final Map<Integer, PartitionSpec> SPECS =
+      ImmutableMap.of(
+          UNPARTITIONED_SPEC.specId(),
+          UNPARTITIONED_SPEC,
+          BY_DATA_SPEC.specId(),
+          BY_DATA_SPEC,
+          BY_DATA_CATEGORY_BUCKET_SPEC.specId(),
+          BY_DATA_CATEGORY_BUCKET_SPEC);
+
+  @Test
+  public void testEmptyMap() {
+    PartitionMap<String> map = PartitionMap.create(SPECS);
+    assertThat(map).isEmpty();
+    assertThat(map).hasSize(0);
+    assertThat(map).doesNotContainKey(Pair.of(1, Row.of(1))).doesNotContainValue("value");
+    assertThat(map.values()).isEmpty();
+    assertThat(map.keySet()).isEmpty();
+    assertThat(map.entrySet()).isEmpty();
```

Comment:
Good spot to check equals

Reply:
Added a separate test case to check both empty and non-empty cases.